packages/ics721/src/msg.rs

```diff
@@ -63,6 +63,10 @@ pub enum CallbackMsg {
         /// Information about the vouchers been redeemed.
         redeem: VoucherRedemption,
     },
+    /// Redeem all entries in outgoing channel.
+    RedeemOutgoingChannelEntries(Vec<(ClassId, TokenId)>),
+    /// Save all entries in incoming channel.
+    AddIncomingChannelEntries(Vec<((ClassId, TokenId), String)>),
```

Comment:
NEW: handle state changes for incoming and outgoing channel in separate callback message